src/modules/map/components/openlayers/utils/print-area-renderer.composable.js

```diff
@@ -0,0 +1,207 @@
+        while (deregister.length > 0) {
+            var item = deregister.pop()
+            if (typeof item === 'function') {
+                item()
+            } else {
+                item.target.un(item.type, item.listener)
+            }
+        }
+        refreshComp()
+    }
+
+    function refreshComp() {
+        updatePrintRectanglePixels(selectedScale)
+        map.render()
+    }
+
+    function updatePrintRectanglePixels(scale) {
+        if (isActive.value) {
+            printRectangle = calculatePageBoundsPixels(scale)
+            map.render()
+        }
+    }
+
+    function calculatePageBoundsPixels(scale) {
+        var s = parseFloat(scale.value)
+        var size = store.getters.mapSize
+        var view = map.getView()
+        var resolution = view.getResolution()
+        var w =
+            (((((size.width / POINTS_PER_INCH) * MM_PER_INCHES) / 1000.0) * s) / resolution) *
+            olHas.DEVICE_PIXEL_RATIO
+        var h =
+            (((((size.height / POINTS_PER_INCH) * MM_PER_INCHES) / 1000.0) * s) / resolution) *
+            olHas.DEVICE_PIXEL_RATIO
+        var mapSize = map.getSize()
+        var center = [
+            (mapSize[0] * olHas.DEVICE_PIXEL_RATIO) / 2,
+            (mapSize[1] * olHas.DEVICE_PIXEL_RATIO) / 2,
+        ]
+
+        var minx, miny, maxx, maxy
+
+        minx = center[0] - w / 2
+        miny = center[1] - h / 2
+        maxx = center[0] + w / 2
+        maxy = center[1] + h / 2
+        return [minx, miny, maxx, maxy]
+    }
+
+    // Compose events
+    function handlePreRender(event) {
+        var context = event.context
+        context.save()
+    }
+    function handlePostRender(event) {
+        // This is where we draw the print area rectangle using the worldPolygon
+        var context = event.context
+        const size = map.getSize()
+
+        var height = size[1] * olHas.DEVICE_PIXEL_RATIO,
+            width = size[0] * olHas.DEVICE_PIXEL_RATIO
+
+        var minx = printRectangle[0],
+            miny = printRectangle[1],
+            maxx = printRectangle[2],
+            maxy = printRectangle[3]
+
+        context.save()
+
+        context.beginPath()
+        // Outside polygon, must be clockwise
+        context.moveTo(0, 0)
+        context.lineTo(width, 0)
+        context.lineTo(width, height)
+        context.lineTo(0, height)
+        context.lineTo(0, 0)
+        context.closePath()
+
+        // Inner polygon,must be counter-clockwise
+        context.moveTo(minx, miny)
+        context.lineTo(minx, maxy)
+        context.lineTo(maxx, maxy)
+        context.lineTo(maxx, miny)
+        context.lineTo(minx, miny)
+        context.closePath()
+
+        context.fillStyle = 'rgba(0, 5, 25, 0.75)'
+        context.fill()
+
+        context.restore()
+    }
+
+    // Compute the optimal scale based on the map size (layout), resolution,
+    // non-covered map vie (by header and menu tray)
+    function getOptimalScale() {
```

Comment:
that also could become a computed properties

Reply:
I am not sure about this one. The function has several references to the store. If we make it a computed property, it will give unnecessary computation. It's also only called once when a user activates the printArea (opening the print menu).

Or is there any other reason?

Reply:
no the main reason was that it would only be calculated when necessary (meaning on value change).
But if what you describe is correct, meaning there will be more changes to its dependencies / more recomputation as number of calls, let's keep it a function then